tests/gocase/unit/disk/disk_test.go

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package disk
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/apache/incubator-kvrocks/tests/gocase/util"
+	"github.com/go-redis/redis/v9"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDisk(t *testing.T) {
+	srv := util.StartServer(t, map[string]string{
+		"rocksdb.compression":       "no",
+		"rocksdb.write_buffer_size": "1",
+		"rocksdb.block_size":        "100",
+	})
+	defer srv.Close()
+	ctx := context.Background()
+	rdb := srv.NewClient()
+	defer func() { require.NoError(t, rdb.Close()) }()
+	estimationFactor := 0.1
+	t.Run("Disk usage String", func(t *testing.T) {
+		require.NoError(t, rdb.Set(ctx, "stringkey", strings.Repeat("A", 1024), 0).Err())
+		val, err := rdb.Do(ctx, "Disk", "usage", "stringkey").Int()
+		require.NoError(t, err)
+		require.GreaterOrEqual(t, val, int(estimationFactor*1024))
+		require.LessOrEqual(t, val, int(1024/estimationFactor))
+	})
+
+	t.Run("Disk usage Hash", func(t *testing.T) {
+		approximateSize := 0
+		for i := 0; i < 10000; i++ {
+			require.NoError(t, rdb.HSet(ctx, "hashkey", "key"+strconv.Itoa(i), "value"+strconv.Itoa(i)).Err())
+			approximateSize += len("hashkey") + len("key"+strconv.Itoa(i)) + len("value"+strconv.Itoa(i)) + 8
+		}
+		require.NoError(t, rdb.Set(ctx, "hashkey", strings.Repeat("A", 1024), 0).Err())
+		val, err := rdb.Do(ctx, "Disk", "usage", "hashkey").Int()
+		require.NoError(t, err)
+		require.GreaterOrEqual(t, val, int(float64(approximateSize)*estimationFactor))
+		require.LessOrEqual(t, val, int(float64(approximateSize)/estimationFactor))
```

Comment:
I am wondering if this could be narrowed down a bit more. This range currently has a hundred times the gap between the upper and lower bound, which I think is still relatively large. But if we can confirm that rocksdb can only do this at most, it is fine.

Reply:
I think it is fine if rocksdb also has a large range.